src/edu/csus/ecs/pc2/core/standings/ScoreboardUtilites.java

```diff
@@ -188,6 +190,95 @@ public static ScoreboardJsonModel createContestStandingsFromJSON(File jsonFile)
     public static int toInt(String string, int defaultNumber) {
         return Utilities.nullSafeToInt(string, defaultNumber);
     }
-    
-    
+
+    /**
+     * 
+     * @param clientId
+     * @param contest
+     * @return
+     */
+    public static Run[] getRunsForUserDivision(ClientId clientId, IInternalContest contest) {
+
+        String division = getDivision(contest, clientId);
+        
+//        System.out.println("debug 22 getRunsForUserDivision for "+clientId+" div is "+division);
+
+        if (ClientType.Type.TEAM.equals(clientId.getClientType())) {
+
+            List<Run> teamRuns = new ArrayList<Run>();
+            for (Run run : contest.getRuns()) {
+
+                ClientId runClientId = run.getSubmitter();
+
+                if (runClientId.equals(clientId)) {
+                    // add team/client's own runs
+                    teamRuns.add(run);
+                } else {
+                    // add if submitting team in same division
+                    if (matchDivsion(contest, division, run.getSubmitter())) {
+                        teamRuns.add(run);
+//                        System.out.println("debug 22 Added run " + run);
+                    }
+                }
+            }
+
+            return (Run[]) teamRuns.toArray(new Run[teamRuns.size()]);
+            
+        } else {
+            return contest.getRuns();
+        }
+    }
+
+    protected static boolean matchDivsion(IInternalContest contest, String inputDivision, ClientId submitter) {
+        
+          String division = getDivision(contest, submitter);
+          
+//          System.out.println("debug 22 matchDivsion  "+inputDivision+" vs "+division+" for "+submitter);
+          
+          if (inputDivision == null && division == null) {
+              return true;
+          } if (inputDivision == null) {
```

Comment:
The code structure here is very misleading.  It _looks_ like the above line is intended to be an `else if`, but there's no `else` keyword.  As a result, the `if` represented on the above line is executed unconditionally (unless the return above is taken).  I suggest either adding an `else`, or adding a line break so it's clear what is intended.

Reply:
reformatted and pushed